src/data/markdown/docs/02 javascript api/07 k6-experimental/03 websockets.md

```diff
@@ -5,22 +5,43 @@ excerpt: "k6 websockets experimental API"
+This experimental API implements the browser [WebSocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) with additional k6-specific functionalities (cookies, tags, headers and so on).
+
+The main difference between this module and [`k6/ws`](/javascript-api/k6-ws/) is that this module uses a global event loop instead of a local one.
+A global event loop lets a single VU have multiple concurrent connections, which improves performance.
+
+The `WebSocket API` is not fully implemented, and we're working on it, but we believe it's usable for most users. So whether you're writing a new WebSocket test, or currently using the `k6/ws` module, we invite you to give it a try, and report any issues in the project's [issue tracker](https://github.com/grafana/xk6-websockets/). Our midterm is to make this module as part of the k6 core, and the long-term goal to replace the [`k6/ws`](/javascript-api/k6-ws/) module.
+
+<!-- vale off -->
+| Class/Method | Description |
+| ------------ | ----------- |
+| [Params](/javascript-api/k6-experimental/websockets/params/)  | Used for setting various WebSocket connection parameters such as headers, cookie jar, compression, etc. |
+| [WebSocket(url, protocols, params)](/javascript-api/k6-experimental/websockets/websocket) | Constructs a new WebSocket connection. |
+| [WebSocket.close()](/javascript-api/k6-experimental/websockets/websocket/websocket-close) | Close the WebSocket connection. |
+| [WebSocket.ping()](/javascript-api/k6-experimental/websockets/websocket/websocket-ping) | Send a ping. |
+| [WebSocket.send(data)](/javascript-api/k6-experimental/websockets/websocket/websocket-send) | Send string data. |
+| [WebSocket.addEventListener(event, handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-addeventlistener) | Add an event listener on the connection for specific event. |
+| [WebSocket.onmessage(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onmessage) | Set up an event listener on the connection for `message` event |
+| [WebSocket.onerror(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onerror) | Set up an event listener on the connection for `error` event |
+| [WebSocket.onopen(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onopen) | Set up an event listener on the connection for `open` connection event |
+| [WebSocket.onclose(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onclose) | Set up an event listener on the connection for `close` connection event |
+| [WebSocket.onping(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onping) | Set up an event listener on the connection for `ping` event |
+| [WebSocket.onpong(handler)](/javascript-api/k6-experimental/websockets/websocket/websocket-onpong) | Set up an event listener on the connection for `pong` event |
```

Comment:
I still think that describing these as `.on*(handler)` is confusing, since they _are_ the handlers, and not functions that receive a handler as an argument. So I would drop the `handler` argument, and only have the Parameter table on each page if the handler itself takes a parameter (like for `onmessage`). So, I guess for `onmessage` this should be `WebSocket.onmessage(data)` instead(?).

And the description might be "Handler function for the `x` event."(?).

But it's not a big deal.

Reply:
Actually, it's a big deal, I believe. Good catch! :bow: and sorry I missed that in your original comment. 

They are just properties that can be assigned with the handler function value (I mean, even `WebSocket.onmessage(data)` is wrong) going to remove `(handler)` part to avoid confusion.